From 6e7f9531b01eb85db432b3206dca7f5c8c55d2c9 Mon Sep 17 00:00:00 2001
From: Colin Kinloch <colin.kinloch@collabora.com>
Date: Sun, 23 Feb 2025 23:40:08 +0000
Subject: [PATCH] controller: Check model is not NULL on notify::is-active

For some reason with GTK 4.17.4 this causes a NULL dereference on window
close.

Co-authored-by: gnome-mpv <gnome-mpv@nowhere>
---
 src/celluloid-controller.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/celluloid-controller.c b/src/celluloid-controller.c
index a124f57d..4a2fdd6d 100644
--- a/src/celluloid-controller.c
+++ b/src/celluloid-controller.c
@@ -571,6 +571,9 @@ is_active_handler(GObject *gobject, GParamSpec *pspec, gpointer data)
 	CelluloidView *view = CELLULOID_VIEW(gobject);
 	gboolean is_active = TRUE;
 
+	// As of GTK 4.17.4 controller->model may be NULL on window close.
+	g_return_if_fail(controller->model);
+
 	g_object_get(view, "is-active", &is_active, NULL);
 
 	if(!is_active)
